### PR TITLE
quick hack to get scripts evaluating on their own thread

### PR DESCRIPTION
**Pull request by ZappoMan**
This is a quick hack that will move Script evaluation onto the ScriptEngine's thread. 

The goal is to fix these crashes we're seeing with entity scripts. The logic behind why this might fix the issue is as follows: The entity scripts are running in a QScriptEngine on their own thread.But when new scripts are downloaded for new entities, they were getting evaluated on the main application thread, We know that QScriptEngine is not thread safe, and so the evaluate() could be corrupting the engine.

This is definitely not an elegant and correct long term solution. We should be moving all ScriptEngine related work onto the thread of the script. Some of the code that currently lives in EntityTreeRenderer (which is mostly running on the main thread) should be moved into ScriptEngine.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: aa1c9e23397ee0c8a9ba725cc45621ad9efc65f4 available at: https://gist.github.com/13ea20e0a37f570c6206<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5782.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5782.dmg

File: libraries/script-engine/src/ScriptEngine.cpp [hunk 1]
```diff
@@ -541,6 +546,17 @@ void ScriptEngine::evaluate() {
 }
 
 QScriptValue ScriptEngine::evaluate(const QString& program, const QString& fileName, int lineNumber) {
+    if (QThread::currentThread() != thread()) {
+        QScriptValue result;
+        QMetaObject::invokeMethod(this,
```
Comment (jherico) [inline libraries/script-engine/src/ScriptEngine.cpp]: Do we not need to use `Qt::BlockingQueuedConnection` in order to ensure that the return value is valid?

Comment (ZappoMan) [inline libraries/script-engine/src/ScriptEngine.cpp]: ok - will fix

Comment (ZappoMan) [inline libraries/script-engine/src/ScriptEngine.cpp]: the fix isn't so easy. I can work on it... but it's not as simple as adding Qt::BlockingQueuedConnection

**Comment by ZappoMan** [comment]
reworking ScriptEngine to be thread safe in a different PR

**Pull request closed without merging**